handlers/sql_generic_v3/handler/init.rb

```diff
@@ -0,0 +1,221 @@
+    # named @info_values.
+    @info_values = {}
+    REXML::XPath.match(@input_document, '/handler/infos/info').each do |node|
+      @info_values[node.attribute('name').value] = node.text
+    end
+
+    # Retrieve all of the handler configuration and store them in a hash attribute
+    # named @configuration.
+    @configuration = {}
+    REXML::XPath.match(@input_document, '/handler/configuration/config').each do |node|
+      @configuration[node.attribute('name').value] = node.text
+    end
+
+	# Retrieve all of the handler parameters and store them in a hash variable named @parameters.
+    @parameters = {}
+    REXML::XPath.each(@input_document, "/handler/parameters/parameter") do |item|
+      @parameters[item.attributes["name"]] = item.text.to_s.strip
+    end
+
+    @enable_debug_logging = @info_values['enable_debug_logging'] == 'Yes'
+    @accept = :json
+    @content_type = :json
+
+  end
+
+  # Executes the specified SQL query.
+  #
+  # This is a required method that is automatically called by the Kinetic Task
+  # Engine.
+  #
+  # ==== Returns
+  # An Xml formatted String representing the return variable results.
+  def execute
+  template_name = @parameters["template_name"]
+  error_handling = @parameters['error_handling']
+  space_form = @info_values["space_form"]
+  db_server = @info_values["database_server"]
+  db_port = @info_values["database_port"]
+  db_name = @parameters["dbname"]
+  db_username = @info_values['database_username']
+  db_password = @info_values["database_password"]
+  error_message = nil
+  
+
+    #checking to see if the server and port are open
+    #this was put in to test java vm access to my local sql server
+    begin
+      kinetic_api_route = "#{@info_values["kinetic_api_location"]}/datastore/forms/" +
+        "#{space_form}/submissions?index=values[Template Name]" +
+        "&q=values[Template Name]=\"#{template_name}\"&include=values"
+      puts "Calling Kinetic Platform at: #{kinetic_api_route}" if @enable_debug_logging
+      
+      response = RestClient::Request.execute \
+        method: 'GET', \
+        url: kinetic_api_route, \
+        user: @info_values["kinetic_api_username"], \
+        password: @info_values["kinetic_api_password"], \
+        headers: {:content_type => @content_type, :accept => @accept}
+      response_code = response.code
+
+      # The query should be indexed to a unique field.
+      submission = JSON.parse(response)['submissions'][0]
+      if (submission.nil?) 
+        raise "No submissions found with a Template Name of '#{template_name}'"
+      end
+
+      query_template = submission["values"]["SQL Query"]
+      if (query_template.nil? || query_template.strip.empty?) 
+        raise "The query was nil or empty.  Check that the '#{space_form}' form has the SQL Query field."
+      end
+
+      socket = TCPSocket.new(db_server, db_port)
+      puts "Port #{db_port} at server '#{db_server}' is open." if @enable_debug_logging

```

Comment:
I don't think this is helpful. If the server is unavailable it is handled in the rescue blocks below.